Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8972,7 +8974,7 @@ void WebPageProxy::makeStorageSpaceRequest(FrameIdentifier frameID, const String
     MESSAGE_CHECK(m_process, frame);
 
     auto originData = SecurityOriginData::fromDatabaseIdentifier(originIdentifier);
-    if (originData != SecurityOriginData::fromURL(URL { currentURL() })) {
+    if (!originData || !protocolHostAndPortAreEqual(originData->toURL(), URL { currentURL() })) {
```

Comment:
Why is this better than relying on `SecurityOriginData::fromURLIgnoringOpaqueness()`?

Reply:
This came from a previous version when we believe we needed `fromURLIgnoringOpaqueness`. We could use `fromURLIgnoringOpaqueness` in these places, but we also avoid any issues by comparing URLs instead of `SecurityOriginData`s. If constructing and comparing `SecurityOriginData` in these checks is better, then I'll make that change.

Reply:
It would seem like a more minimal change